File: 05_programacion_modular/ejemplos/usa_matematicas.py
"""
usa_matematicas.py — Programa principal que importa nuestro módulo.

Demuestra cómo, una vez creado matematicas.py, podemos USARLO en cualquier
programa con un simple import. El módulo es ahora una caja de herramientas
reutilizable.

Tema 5 - Introducción a la Computación Científica (ICC).
"""

import matematicas as m


def main():
    print("Usando el módulo matematicas.py importado")
    print("=" * 45)

    # Calculamos el MCD y MCM de varios pares
    print("\nMáximo Común Divisor:")
    print(f"  mcd(48, 18) = {m.mcd(48, 18)}")
    print(f"  mcd(1071, 462) = {m.mcd(1071, 462)}")

    # Comprobamos primalidad
    print("\nPrimos en [10, 30]:")
    for n in range(10, 31):
        if m.es_primo(n):
            print(f"  {n} es primo")

    # Calculamos √2 sin importar math.sqrt
    print(f"\n√2 (método babilónico) ≈ {m.raiz_babilonica(2):.15f}")

    # Combinatoria
    print(f"\nC(10, 3) = {m.combinatorio(10, 3)} (esperado: 120)")
    print(f"5! = {m.factorial(5)} (esperado: 120)")


if __name__ == "__main__":
    main()

File: 05_programacion_modular/ejemplos/usa_matematicas.c
/*
 * usa_matematicas.c — Programa principal que usa nuestro módulo en C.
 *
 * Compilar con:
 *   gcc -o usa usa_matematicas.c matematicas.c -lm
 * Ejecutar con:
 *   ./usa
 *
 * Tema 5 - Introducción a la Computación Científica (ICC).
 */

#include <stdio.h>
#include "matematicas.h"


int main(void) {
    printf("Usando el módulo matematicas (C)\n");
    printf("================================\n\n");

    printf("MCD:\n");
    printf("  mcd(48, 18) = %d\n", mcd(48, 18));
    printf("  mcd(1071, 462) = %d\n\n", mcd(1071, 462));

    printf("Primos en [10, 30]:\n");
    for (int n = 10; n <= 30; n++) {
        if (es_primo(n)) printf("  %d es primo\n", n);
    }

    printf("\nFactorial y combinatorio:\n");
    printf("  5! = %ld\n", factorial(5));
    printf("  C(10, 3) = %ld\n", combinatorio(10, 3));

    printf("\nRaíz babilónica:\n");
    printf("  √2 ≈ %.15f\n", raiz_babilonica(2.0));

    return 0;
}
